v2/internal/testcommon/vcr/recorder.go

```diff
@@ -26,6 +27,9 @@ type Interface interface {
 	// IDs returns the available Azure resource IDs for the test
 	IDs() creds.AzureIDs
 
+	// Redactor returns the redactor associated with the recorder
+	Redactor() *Redactor
```

Comment:
Instead of providing direct access to the redactor (which is kinda an implementation detail, albeit an important one), how about changing this method to add a redaction?

```suggestion
	AddRedaction(redactionValue string, replacementValue string)
```